### PR TITLE
Increase PHPStan to level 6

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for next major.

## Changelog

```markdown
### Changed
- Change `SnapshotChildrenCollection` to be a lazy `Collection` from Doctrine
```

### CODE REVIEW

File: src/Model/Template.php [hunk 1]
```diff
@@ -55,18 +76,20 @@ public function getContainers()
...
     }
 
     /**
-     * @param $code
+     * @param string $code
+     *
+     * @return array<string, mixed>
      *
-     * @return array
+     * @phpstan-return Container|array{}
```
Comment (VincentLanglet) [inline src/Model/Template.php]: Seems weird to return `array{}` if no container was found.

It should be a LogicException (or at least null).

Comment (jordisala1991) [inline src/Model/Template.php]:
```suggestion
     * @phpstan-return Container|null
```

File: src/Model/Template.php [hunk 2]
```diff
@@ -55,18 +76,20 @@ public function getContainers()
...
     public function addContainer($code, $meta): void
     {
         $this->containers[$code] = $this->normalize($meta);
     }
 
     /**
-     * @param $code
+     * @param string $code
+     *
+     * @return array<string, mixed>
```
Comment (jordisala1991) [inline src/Model/Template.php]:
```suggestion
     * @return array<string, mixed>|null
```

File: src/Model/Template.php (hunk 1 above)
Comment (jordisala1991) [reply src/Model/Template.php]: I went for null instead of exceptions

**Review by VincentLanglet — APPROVED** [review]

File: src/Model/Template.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Model/Template.php]: It makes sens since there is no has method

**Pull request merged**